File: Tram ride/Tram_Ride.java
import java.util.*;

public class TramRide {
    public static int solve(int N, int start, int finish, int[] ticketCost) {
        int totalCost = 0;

        if (start == finish) {
            return totalCost;
        }

        if (start > finish) {
            int temp = start;
            start = finish;
            finish = temp;
        }

        for (int i = start; i < finish; i++) {
            totalCost += ticketCost[i];
        }

        int reverseCost = 0;
        for (int i = finish - 1; i >= start; i--) {
            reverseCost += ticketCost[i];
        }

        return Math.min(totalCost, reverseCost);
    }

    public static void main(String[] args) {
        int N = 4;
        int start = 1;
        int finish = 4;
        int[] ticketCost = {1, 2, 2, 4};

        int minCost = solve(N, start - 1, finish - 1, ticketCost);

        System.out.println(minCost); // Output: 4
    }
}

File: Tram ride/Tram-Ride.py
def solve(N, start, finish, ticketCost):
    totalCost = 0
    
    if start == finish:
        return totalCost
    
    if start > finish:
        start, finish = finish, start
    
    for i in range(start, finish):
        totalCost += ticketCost[i]
    
    reverseCost = 0
    for i in range(finish - 1, start - 1, -1):
        reverseCost += ticketCost[i]
    
    return min(totalCost, reverseCost)
    
N = 4
start = 1
finish = 4
ticketCost = [1, 2, 2, 4]

minCost = solve(N, start - 1, finish - 1, ticketCost)

print(minCost) # Output: 4
